### DISCUSSION TITLE
Merkle Root check failed.

### DISCUSSION THREAD

**Discussion by AmitKawad** [ConsenSys/quorum#1629] [Q&A]
Setup Details:
We are running GoQuorum with the high availability setup over AKS.

**Quorum Version:** 22.7.4
**Number of Parties**: 2 (say Party A and Party B)
**Quorum:** 3 pods for both parties

**Scenario**:
We have observed multiple occurrences of **merkle root check failed** in quorum logs on both the parties as mentioned in the above setup details. We have also encountered some transactions failed with status **0x0** around this time. Does this lead to merkle root check failed error in the quorum logs?

Below is one of the block where status **0x0** was seen. 
![image](https://user-images.githubusercontent.com/37235516/226913454-deda74c9-1c67-4939-be3d-44cccc1db04b.png)

**Note**: The merkle root check failed error log has not occurred always on the same pod. It has been observed on different pods in different occurrences. 

**Sample log from quorum pod** : _Merkle root check failed actual=3f8d3b..886510 expect=787840..74d076_

**How do we validate that the chain is not corrupted or auto-repaired after we get this error. As we have not seen any syncing issues**

**Comment by antonydenyer** [comment]
This can happen if the db hasn't got the private transaction at the point the member validates the onchain transaction. You can work round it using.

https://docs.goquorum.consensys.net/configure-and-manage/manage/enhanced-permissions

**Reply by AmitKawad** [reply]
Hi @antonydenyer,
We are observing this as a debug log 
**_DEBUG_**[03-17|10:08:51.012] Merkle Root check failed actual=3cd6ef..b68328 expect=e206b8..2f1e1d 
Can this cause any potential data corruption/bad block issue at later point of time? Shall we treat it as an error or just an info/debug log?

**Reply by antonydenyer** [reply]
In that scenario the sending transaction should fail, because the state root has changed (and you're using privacy enhancements) between the point of submission and the point of verification.

**Reply by AmitKawad** [reply]
Hi @antonydenyer Thanks for the reply, 
For transaction failure we are re-posting the transactions. Does it have any impact on future transactions/contract state. If it doesn't affect the contract state, is it safe to ignore the debug log as we are re-posting the transactions? Please suggest.

**Reply by antonydenyer** [reply]
It's safe to ignore the debug message in that scenario yes

**Reply by ghost** [reply]
Hi @antonydenyer, can you please suggest on #1631 (What are the logs to be monitored to identify/avoid GoQuorum's Data Corruption and Bad Block Issues? Can we get a list of error/debug logs)?

**Reply by ghost** [reply]
> Hi @antonydenyer, can you please suggest on #1631 (What are the logs to be monitored to identify/avoid GoQuorum's Data Corruption and Bad Block Issues? Can we get a list of error/debug logs)?

Hi @antonydenyer, can you please suggest us on this?
